File: workflow/scripts/bed2saf.R
saf <- read.table(snakemake@input[[1]])
colnames(saf) <- c("Chr", "Start", "End", "GeneID", "frame", "Strand")
saf <- saf[, c("GeneID", "Chr", "Start", "End", "Strand")]
write.table(saf, snakemake@output[[1]], sep = "\t", col.names = TRUE, row.names = FALSE)

File: workflow/scripts/bed2saf.py
#!/usr/bin/env python
import pandas as pd

bed = pd.read_csv(snakemake.input[0], sep='\t')
bed.columns = ["Chr", "Start", "End", "GeneID", "frame", "Strand"]
bed.to_csv(snakemake.output[0], sep='\t',columns=["GeneID", "Chr", "Start", "End", "Strand"],index=False)
